src/main/resources/assets/paths/datasetV3ById.yaml

```diff
@@ -0,0 +1,46 @@
+put:
+  summary: Updates a dataset from a valid DatasetUpdate object
+  description: Updates a dataset from a valid DatasetUpdate object
+  parameters:
+    - name: datasetId
+      in: path
+      description: The Dataset ID
+      required: true
+      schema:
+        type: integer
+  requestBody:
+    description: Returns the updated dataset. A dataset must already exist, but files can be either
+                 updated or created if they don't already exist.
+    content:
+      multipart/form-data:
+        schema:
+          type: object
+          properties:
+            dataset:
+              description: uses DatasetUpdate record class which has the values and properties of the 
+                           dataset that can be updated. The purpose of this class is to allow for 
+                           specific adjustments for this endpoint.
+              type: object
+            alternativeDataSharingPlan:
+              description: Optional alternative data sharing plan
+              type: string
+              format: binary
+            consentGroups[0].nihInstitutionalCertificationFile:
+              description: Certification file
+              type: string
+              format: binary
+  tags:
+    - Dataset
+  responses:
+    201:
+      description: Successfully updated Datasets
+      content:
+        application/json:
+          schema:
+            $ref: '../schemas/DatasetUpdate.yaml'
```

Comment:
We're not returning a dataset update object, we're returning a full dataset object. However, we do need to somehow express that the values in `DatasetUpdate.yaml` are what to use for input. Since we can't use `$ref` where we want to, we should move all of the content in `DatasetUpdate.yaml` into the description and then reference `Dataset.yaml` here on line 40.